src/sync/rwlock.rs

```diff
@@ -147,11 +147,15 @@ impl<T: ?Sized> RwLock<T> {
         // Block if the lock is in a state where we can't acquire it immediately
         match &state.holder {
             RwLockHolder::Write(writer) => {
-                assert_ne!(*writer, me);
+                if *writer == me {
+                    panic!("deadlock! task {:?} tried to acquire a RwLock it already holds", me);
+                }
```

Comment:
nit: Should be *an* `RwLock`, like in the [doc](https://doc.rust-lang.org/std/sync/struct.RwLock.html). Same below.